src/world/Spell/Spell.Legacy.cpp

```diff
@@ -926,17 +926,16 @@ uint8 Spell::prepare(SpellCastTargets* targets)
 
     if (objmgr.IsSpellDisabled(GetSpellInfo()->getId()))//if it's disabled it will not be casted, even if it's triggered.
         cancastresult = uint8(m_triggeredSpell ? SPELL_FAILED_DONT_REPORT : SPELL_FAILED_SPELL_UNAVAILABLE);
-    else if (m_triggeredSpell || ProcedOnSpell != nullptr)
-        cancastresult = SPELL_CANCAST_OK;
     else
-        cancastresult = CanCast(false);
+        cancastresult = canCast(false);
 
     LogDebugFlag(LF_SPELL, "CanCast result: %u. Refer to SpellFailure.h to work out why." , cancastresult);
 
     ccr = cancastresult;
     if (cancastresult != SPELL_CANCAST_OK)
     {
-        SendCastResult(cancastresult);
+        // Triggered spells also need to go through cancast check but they do not pop a error message
```

Comment:
Why? What incorrect behaviour is caused by skipping this check?

Reply:
Nothing is skipped? Triggered spells also need to be checked in canCast() (for range, aurastate etc) but they just do not send error message to client. Before this change triggered spells weren't even checked in canCast(). See https://github.com/AscEmu/AscEmu/pull/572/files#diff-a5339c3ab8259ee5d6d4a3a957d11f80L929

Reply:
@Appled That's what I'm referring to.

Previously, these checks were skipped for triggered spells, however now you're wanting to apply them.

Are you certain that's the correct behaviour?

Reply:
@Evairfairy Oh I've managed to misunderstand your question then. There are certain things which must be checked in canCast (which are not yet implemented), such as death state for triggered spells which didn't proc from an aura, target checks, power checks etc. However legacy canCast is not called for triggered spells, see here https://github.com/Appled/AscEmu/blob/9168eb6983f714d60f68dc8fa302928bb3ecc4ee/src/world/Spell/Spell.cpp#L95